src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -105,16 +104,67 @@ protected override void OnNextCore(Block block)
         }
 
         /// <summary>
-        /// A loop method that continuously relays blocks found in <see cref="blocksToAnnounce"/> to connected peers on the network.
+        /// Continuously dequeues items from <see cref="blocksToAnnounce"/> and sends
+        /// them  to the peers after the timer runs out or if the last item is a tip.
+        /// </summary>
+        private async Task DequeueContinuouslyAsync()
+        {
+            var batch = new List<ChainedBlock>();
+
+            Task<ChainedBlock> dequeueTask = null;
+            Task timerTask = null;
```

Comment:
Too many tasks flying around its starting to make me feel uncomfortable.

Reply:
Less then 2 at a time 🙂

Reply:
@dangershony actually i was first opposed to this solution as well, but we worked on that and it allowed us to have a lock free implementation, which is short enough that one can argue about it and check all its states, so this is actually pretty good result and i'm happy @noescape00 was persistent enough here to convince me we can do reasonably well with all those tasks around